src/main/java/org/broadinstitute/hellbender/utils/test/IntegerReadCountFileComparator.java

```diff
@@ -0,0 +1,112 @@
+package org.broadinstitute.hellbender.utils.test;
+
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.tools.copynumber.coverage.readcount.ReadCountFileHeaderKey;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.testng.Assert;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.IntStream;
+
+/**
+ * This is a testing utility class to check the correctness of coverage collection results for CNV analysis.
+ * It assumes that results come in a TSV format, and checks for exact string matches between two tables. Therefore,
+ * this class is NOT intended to be used with output formats containing floats, as it doesn't allow for rounding error.
+ */
+public class IntegerReadCountFileComparator {
+    private final List<String[]> values;
+    private final int columnCount;
+    private final int rowCount;
+    private final String[] columnNames;
+    private final Map<String, String> headerValuesMap;
+
+    private final static String TAB_SEPARATOR_REGEX = "\\t";
+    private final static String COMMENT_PREFIX_START_REGEX = "^#.*$";
+
+    /**
+     * Private constructor
+     *
+     * @param file input file
+     * @throws IOException if there is any problems while reading the file
+     * @throws UserException.BadInput if non-integer value is encountered below column names line or file
+     * is not formatted correctly
+     */
+    private IntegerReadCountFileComparator(final File file) throws IOException {
+        Utils.nonNull(file, "the file cannot be null");
```

Comment:
Can you remove all non-null messages?  I think they are overkill, and it takes effort to make sure they are formatted consistently.